### PR TITLE
Add a function to generate random strings

### PR DESCRIPTION
**Pull request by hrpatel**

### CODE REVIEW

File: shtdlib.sh [hunk 1]
```diff
@@ -1654,6 +1654,17 @@ function ln_sf {
     debug 10 "Successfully created symlink"
 }
 
+# Create string of random characters
+#  - First param is length, default: 20
+#  - Second param is characters, default: A-Za-z0-9_
```
Comment (antevens) [inline shtdlib.sh]: I'd clarify that dash is not one of the default characters.

File: shtdlib.sh [hunk 2]
```diff
@@ -1654,6 +1654,17 @@ function ln_sf {
...
+gen_rand_chars() {
+    local length="${1:-20}"
+    local chars="${2:-A-Za-z0-9_}"
+    debug 10 "Creating a string of random characters of length ${length}"
+    LC_CTYPE=C tr -dc ${chars} < /dev/urandom | head -c ${length}
+}
+
```
Comment (antevens) [inline shtdlib.sh]: Should probably just be one line

File: shtdlib.sh (hunk 1 above)
Comment (hrpatel) [reply shtdlib.sh]: Done

File: shtdlib.sh (hunk 2 above)
Comment (hrpatel) [reply shtdlib.sh]: Yeah, I thought about it but having it like this gives us the `debug` message which could come in handy with verbose logging

**Pull request merged**